### ISSUE TITLE
Retour d'utilisation : 

### ISSUE THREAD

**Issue by grillead** [grillead/freeboxandroidhomecontrol#1]
Merci a ceux qui utilise la vm et le script de me faire un petit retour afin de pouvoir faire si possible des modif, je l utilise actuellement chez moi avec une config serveur delta + player pop.

**Comment by sebastienheyd** [comment]
Hello,

Joli boulot ! J'ai juste une remarque, quand l'assistant zappe ça revient à chaque fois sur la "home" de Oqee avant de zapper.

Il faudrait, je pense, vérifier que l'intent courante n'est pas déjà "net.oqee.androidtv", si oui on change juste la chaine avec le input text, si non, on charge Oqee.

Jouable je pense

**Comment by grillead** [comment]
> Hello,
> 
> Joli boulot ! J'ai juste une remarque, quand l'assistant zappe ça revient à chaque fois sur la "home" de Oqee avant de zapper.
> 
> Il faudrait, je pense, vérifier que l'intent courante n'est pas déjà "net.oqee.androidtv", si oui on change juste la chaine avec le input text, si non, on charge Oqee.
> 
> Jouable je pense

Merci 👍 
Oui il faudrais que je regarde si possible via adb d avoir l info que oqee et bien lancé au 1er plan pour lancé la chaine, je vais regarder ça a l occaz, si je trouve la soluce je mettrais a jour, et il faudra juste reboot la vm (normalement j ai un script d update au boot pour mettre à jour par rapport au github ;) )

**Comment by grillead** [comment]
> Il faudrait, je pense, vérifier que l'intent courante n'est pas déjà "net.oqee.androidtv", si oui on change juste la chaine avec le input text, si non, on charge Oqee.

Solution trouvée, j ai fait le test chez moi ce soir (depuis netflix ou amazon prime,  la box bascule sur Oqee, si Oqee au 1er plan elle envoie juste le numeros de chaine).
Commit effectué sur le GIT, un reboot de la vm ou un lancement du script ./update.sh depuis la console et le tour et joué ^^ sur ce, je vais me coucher 😴

**Comment by sebastienheyd** [comment]
Ça fonctionne ! Nickel !

**Comment by grillead** [comment]
> Ça fonctionne ! Nickel !

Super, merci de ton retour !

**Comment by thunderj27** [comment]
Bonjour @grillead , 
Super job! 
Je fais une install avec nginx & php 7. Petite migration vers PDO au passage.
Je ne saisis pas bien l'intérêt de $cmd dans ton script étant donné que IFTTT envoie que via ?nom. Il y a t'il quelque chose qui m'échappe?
Concernant ton dernier commit, cela ne semble pas fonctionner chez moi :/. Il faut que je regarde cela de plus prêt.

**Comment by grillead** [comment]
@thunderj27 Merci de ton retour, le $cmd est une attente non terminée :) j avais dans l idée de pouvoir lancer des app via google, mais ne pouvant les controler ensuite à la voix cela pert un peu de son interet ^^
Pour le derniers commit (du freebox.php) effectivement il ne marche plus non plus chez moi depuis peu, il faut que je me repenche dessus un soir, suite au dernieres maj free il n est plus efficace ( il evitait le retour a l interface oqee quand celle ci etait deja lancé).

PS : j avoue que j'ai fait avec mes maigres/ancienne connaisance concernant php (j en ai fait un peu y a 18ans au lycée du coup je suis resté sur mes acquis xD , pis comme disent les vieux , c 'etait mieux avant ^^ )

**Comment by grillead** [comment]
@thunderj27 J ai trouvé le soucis patché, je te laisse jeter un oeil au dernier commit ;) une simple histoire de -| +; et de -0 +1

**Comment by thunderj27** [comment]
@grillead : Bien vu :). Du coup je suis passé par un test du résultat du grep | grep et cela fonctionne aussi.

**Comment by AlexandreGicquel** [comment]
Salut,
Je n'arrive pas à voir accès à ta VM, le lien ne donne rien.
Est-ce normal ?
Merci.

**Comment by freemans32** [comment]
Bonjour je n'arrive pas à faire fonctionner la chose, pourriez vous détailler le webhook ? y'a t'il un moyen de tester en interne depuis la console ou autre un envoi de commande a la freebox pop ?

**Comment by freemans32** [comment]
est ce qu'il y a un log sur le serveur pour voir si les commandes sont bien reçu ?

**Comment by grillead** [comment]
@freemans32 Bonjour, 
Pour tester tu peut le faire depuis n importe quelle navigateur web d un pc ou télephone connecté au reseau, dans le cas chez moi par exemple : http://192.168.0.30:1122/freebox.php?nom=Nickelodeon%20Junior (192.168.0.30 etant l ip de ma vm)

Oui il y a des log sur le serveur, il faut se connecter en ssh dessus (avec putty par exemple) le log se trouve dans le dossier log apache, tu peut faire via ssh un tail -f /var/log/apache2/access.log afin d avoir le fichier log en live depuis la console.

**Comment by freemans32** [comment]
> @freemans32 Bonjour,
> Pour tester tu peut le faire depuis n importe quelle navigateur web d un pc ou télephone connecté au reseau, dans le cas chez moi par exemple : http://192.168.0.30:1122/freebox.php?nom=Nickelodeon%20Junior (192.168.0.30 etant l ip de ma vm)
> 
> Oui il y a des log sur le serveur, il faut se connecter en ssh dessus (avec putty par exemple) le log se trouve dans le dossier log apache, tu peut faire via ssh un tail -f /var/log/apache2/access.log afin d avoir le fichier log en live depuis la console.

Effectivement ça fonctionne avec ton lien, il faut absolument dire le nom de la chaîne ? On peut le faire aussi avec le numéro de la chaîne ou pas ? J'ai essayé avec la chaîne 207 :http://192.168.0.30:1122/freebox.php?nom=207 mais ça n'a pas l'air de fonctionner c'est normal ?

**Comment by grillead** [comment]
> > @freemans32 Bonjour,
> > Pour tester tu peut le faire depuis n importe quelle navigateur web d un pc ou télephone connecté au reseau, dans le cas chez moi par exemple : http://192.168.0.30:1122/freebox.php?nom=Nickelodeon%20Junior (192.168.0.30 etant l ip de ma vm)
> > Oui il y a des log sur le serveur, il faut se connecter en ssh dessus (avec putty par exemple) le log se trouve dans le dossier log apache, tu peut faire via ssh un tail -f /var/log/apache2/access.log afin d avoir le fichier log en live depuis la console.
> 
> Effectivement ça fonctionne avec ton lien, il faut absolument dire le nom de la chaîne ? On peut le faire aussi avec le numéro de la chaîne ou pas ? J'ai essayé avec la chaîne 207 :http://192.168.0.30:1122/freebox.php?nom=207 mais ça n'a pas l'air de fonctionner c'est normal ?

Oui pour zappé sur une chaine via son numeros : il faut remplacer le freebox.php?nom= par freebox.php?cmd= 
il faut aussi modifier dans ifft et ne mettre en IF : "Say a phrase with a number" ex : zappe sur la chaine # 
webhook : http://ip_externe_box:1122/freebox.php?cmd={{NumberField}}

**Comment by freemans32** [comment]
> > > @freemans32 Bonjour,
> > > Pour tester tu peut le faire depuis n importe quelle navigateur web d un pc ou télephone connecté au reseau, dans le cas chez moi par exemple : http://192.168.0.30:1122/freebox.php?nom=Nickelodeon%20Junior (192.168.0.30 etant l ip de ma vm)
> > > Oui il y a des log sur le serveur, il faut se connecter en ssh dessus (avec putty par exemple) le log se trouve dans le dossier log apache, tu peut faire via ssh un tail -f /var/log/apache2/access.log afin d avoir le fichier log en live depuis la console.
> > 
> > 
> > Effectivement ça fonctionne avec ton lien, il faut absolument dire le nom de la chaîne ? On peut le faire aussi avec le numéro de la chaîne ou pas ? J'ai essayé avec la chaîne 207 :http://192.168.0.30:1122/freebox.php?nom=207 mais ça n'a pas l'air de fonctionner c'est normal ?
> 
> Oui pour zappé sur une chaine via son numeros : il faut remplacer le freebox.php?nom= par freebox.php?cmd=
> il faut aussi modifier dans ifft et ne mettre en IF : "Say a phrase with a number" ex : zappe sur la chaine #
> webhook : http://ip_externe_box:1122/freebox.php?cmd={{NumberField}}

Ok merci, y'a la possibilité de faire les 2 avec both a number and text ingredient ?

**Comment by freemans32** [comment]
Je viens de tester avec cmd=2 et ca ne fonctionne pas, je vois bien apparaître la commande dans la log mais ca change pas de chaine que ce soit en interne ou externe, quand je le fais en interne depuis un navigateur avec nom=france 2  j'obtiens ça:
`france 2
uuid-webtv-201
2` 
mais avec cmd=2 rien ne s'affiche

**Comment by grillead** [comment]
> Je viens de tester avec cmd=2 et ca ne fonctionne pas, je vois bien apparaître la commande dans la log mais ca change pas de chaine que ce soit en interne ou externe, quand je le fais en interne depuis un navigateur avec nom=france 2  j'obtiens ça:
> `france 2
> uuid-webtv-201
> 2` 
> mais avec cmd=2 rien ne s'affiche

Ha, il faut que je regarde j ai du faire une boulette dans le code avec ma dernière update, je vais regarder des que j'ai un moment, je ferais la modif il suffira juste de faire un reboot de la vm pour mettre a jour, je te tiens au courant quand c est résolu

**Comment by grillead** [comment]
> > Je viens de tester avec cmd=2 et ca ne fonctionne pas, je vois bien apparaître la commande dans la log mais ca change pas de chaine que ce soit en interne ou externe, quand je le fais en interne depuis un navigateur avec nom=france 2  j'obtiens ça:
> > `france 2 uuid-webtv-201 2`
> > mais avec cmd=2 rien ne s'affiche
> 
> Ha, il faut que je regarde j ai du faire une boulette dans le code avec ma dernière update, je vais regarder des que j'ai un moment, je ferais la modif il suffira juste de faire un reboot de la vm pour mettre a jour, je te tiens au courant quand c est résolu

Bug résolu, un simple reboot de la vm ou via ssh : sudo ./update et le changement via le numéros sera ok ;)

**Comment by freemans32** [comment]
> > > Je viens de tester avec cmd=2 et ca ne fonctionne pas, je vois bien apparaître la commande dans la log mais ca change pas de chaine que ce soit en interne ou externe, quand je le fais en interne depuis un navigateur avec nom=france 2  j'obtiens ça:
> > > `france 2 uuid-webtv-201 2`
> > > mais avec cmd=2 rien ne s'affiche
> > 
> > 
> > Ha, il faut que je regarde j ai du faire une boulette dans le code avec ma dernière update, je vais regarder des que j'ai un moment, je ferais la modif il suffira juste de faire un reboot de la vm pour mettre a jour, je te tiens au courant quand c est résolu
> 
> Bug résolu, un simple reboot de la vm ou via ssh : sudo ./update et le changement via le numéros sera ok ;)

Merci pour ton retour rapide et ta résolution, y'a la possibilité de faire les 2 avec both a number and text ingredient ?

**Comment by thunderj27** [comment]
@freemans32 & @grillead j'ai remarqué que IFTTT renvoie parfois des chiffres et cela même avec un text ingredient. De mon côté, j'ai fait quelques modifs pour que le nom de la chaîne ou le numéro puissent effectuer l'action. Je voulais partager tout ceci mais je n'ai pas trouvé le temps jusqu'à maintenant. Je vais voir cela dès que possible si ca peut aider. Bonne soirée

**Comment by grillead** [comment]
> 
> Merci pour ton retour rapide et ta résolution, y'a la possibilité de faire les 2 avec both a number and text ingredient ?

Tu peut le faire avec 2 applet ifttt 
1 avec numbers et l'autre avec text (zappe sur la chaine numéros # / zappe sur la chaine $)
Le both number AND text indique que tu vas envoyer une commande avec du texte ET des chiffres et pas du texte ET/OU des chiffres donc ça me semble pas possible

**Comment by CR95580** [comment]
@grillead 
Je voudrais installer ta solution de commande du player Free. Avant de me lancer, j'ai testé ma configuration avec ADB (seul) mais je ne me connecte pas. La commande nmap me retourne que les ports 8008, 8009, 8443, 9000 et 9080 sont ouverts mais pas le port 5555. Idem avec adb connect -> connection refused. J'en déduis que le port 5555 n'est pas ouvert. J'ai bien fait la manip de 7 appuis sur le bouton build et j'ai activé "Options pour les développeurs"pour avoir l'accès développeur. Par contre, j'ai lu ailleurs qu'il fallait mettre l'accès développeur en mode réseau en tapant la commande adb TCPIP 5555 alors que la box est branchée sur un câble USB (sinon elle reste en mode USB) et ensuite débrancher le câble USB.
Est-ce que tu as fait cette opération la 1ére fois ?
Dans ce cas, il faut un câble USB avec 2 connecteurs larges pour connecter le PC et le Player ?  Je n'ai que des câbles avec un connecteur large coté PC et un petit connecteur USB pour se brancher du coté téléphone.
Merci de ton retour

**Comment by grillead** [comment]
> @grillead 
> Je voudrais installer ta solution de commande du player Free. Avant de me lancer, j'ai testé ma configuration avec ADB (seul) mais je ne me connecte pas. La commande nmap me retourne que les ports 8008, 8009, 8443, 9000 et 9080 sont ouverts mais pas le port 5555. Idem avec adb connect -> connection refused. J'en déduis que le port 5555 n'est pas ouvert. J'ai bien fait la manip de 7 appuis sur le bouton build et j'ai activé "Options pour les développeurs"pour avoir l'accès développeur. Par contre, j'ai lu ailleurs qu'il fallait mettre l'accès développeur en mode réseau en tapant la commande adb TCPIP 5555 alors que la box est branchée sur un câble USB (sinon elle reste en mode USB) et ensuite débrancher le câble USB.
> Est-ce que tu as fait cette opération la 1ére fois ?
> Dans ce cas, il faut un câble USB avec 2 connecteurs larges pour connecter le PC et le Player ?  Je n'ai que des câbles avec un connecteur large coté PC et un petit connecteur USB pour se brancher du coté téléphone.
> Merci de ton retour

Salut, as-tu pensez a activer le débogage usb dans les options developper ? ;-) pour ma part je n ai jamais eu besoin de la connecter en usb, la simple activation de l option debogage usb m as permis la connexion adb via le reseau.
Tiens moi au courant ^^

**Comment by CR95580** [comment]
Effectivement, l'activation debogage usb donne l'accès au player (avec la mention unauthorized). 
Merci.

**Comment by grillead** [comment]
> Effectivement, l'activation debogage usb donne l'accès au player (avec la mention unauthorized).
> Merci.

;-) le unauthorized vient du fais qu il faut autoriser une premiere fois le client adb (message qui s affiche sur l écran) une fois valider et enregistrer tout vas bien :)

**Comment by CR95580** [comment]
@grillead
Oui, ça marche, surtout quand j'ai compris qu'il faut être devant son PC et en même temps devant la télé !
Est-ce que tu sais s'il y a une commande pour "réveiller" la POP ?
Plus généralement, ou peut-on trouver les commandes que l'on peut envoyer à la Pop comme "Input keyevent..." ou "monkey..." ?
Merci

**Comment by freemans32** [comment]
Moi pour l'allumer j'utilise la commande "allume la freebox" du chromecast de la freebox, par contre il faut désactiver la veille prolongée.

**Comment by grillead** [comment]
> @grillead
> Oui, ça marche, surtout quand j'ai compris qu'il faut être devant son PC et en même temps devant la télé !
> Est-ce que tu sais s'il y a une commande pour "réveiller" la POP ?
> Plus généralement, ou peut-on trouver les commandes que l'on peut envoyer à la Pop comme "Input keyevent..." ou "monkey..." ?
> Merci

Pour le reveille, j utilise aussi la fonction de reveil via chromecast "allume la freebox" si jamais google indique que la freebox n est pas accessible il faut désactivé la veille profonde ( Paramètres > Préférence relative à l'appareil > Paramètres avancés > Economie d'energie > Veille profonde : Désactivé )
Pour les commandes je les avais glané sur developer.android.com il s agit des commandes adb shell .

**Comment by CR95580** [comment]
Merci pour les conseils

J'ai désactivé la veille profonde et le Player reste allumé en permanence : c'est normal ?
Dans ce cas, il n'y a plus besoin de le "réveiller" ?

**Comment by grillead** [comment]
@CR95580 Normalement ton player se mets en veille classique via la commande ok google eteins la freebox ou via la bouton veille de la telecommande.
Concernant la mise en veille tu peut aussi activer la mise en veille auto après 2h / 4H... d'inactivitée.
J ai aussi mis a jour les fichiers de controle, tu peut mettre a jour ta VM en lui faisant 2 reboot ou en lancant manuellement via ssh le script d update 2 fois (sudo ./update.sh) une fois a jour avec les derniers update la vm se mettra a jour automatiquement apres chaque reboot ;)

**Comment by thunderj27** [comment]
Bonjour @grillead,
Beaucoup de temps et de choses se sont passées de mon côté depuis 2021... Je me demandais, suite au changemenent "de politique" de Google et IFTTT, as tu trouvé des alternatives? 
Cordialement,

**Comment by grillead** [comment]
> Bonjour @grillead, Beaucoup de temps et de choses se sont passées de mon côté depuis 2021... Je me demandais, suite au changemenent "de politique" de Google et IFTTT, as tu trouvé des alternatives? Cordialement,

Salut, désolé de ma réponse tardive, pour ma part je suis passer sur la solution cloud kodono d Aymeric( https://github.com/Aymkdn/assistant-freebox-cloud) , je ne sais pas si d'autre service propose l équivalent d ifttt, apres je sais qu il est possible de faire des routine en code pour google assistant, a voir si possible de gerer des variable à envoyé a la vm peut être ...
